colossalai/tensor/graph/graph_node.py

```diff
@@ -0,0 +1,92 @@
+from colossalai.tensor import ColoTensor
+from colossalai.context.singleton_meta import SingletonMeta
+
+
+class GraphGlobalEnv(metaclass=SingletonMeta):
+
+    def __init__(self) -> None:
+        self.graph_building = False
+        self.graph_node_list = []
+
+    def add_graph_node(self, node):
+        self.graph_node_list.append(node)
+
+
+class GraphContext():
+    """
+    
+    Building the computing graph under the context
+
+    >>> with GraphContext():
+    >>>     output = model(colo_input_tensor)
+    """
+    graph_nodes = []
+
+    def __enter__(self):
+        GraphGlobalEnv().graph_building = True
+
+    def __exit__(self, *exc_info):
+        GraphGlobalEnv().graph_building = False
+        self.graph_nodes = GraphGlobalEnv().graph_node_list
+
+
+class GraphNode(object):
+    myid = 0
```

Comment:
We need a func to reset this global var before running a model every time.

Reply:
We just run the graph building once under GraphContext.

Reply:
Also, what about using param name itself as node identification.

Reply:
OK, the name can be an attribute.
I have updated the id resetting.

Reply:
The name can be very long in some cases. Now just use a number.